From 38de880b4d939fb81da254fe99dc501bbaa80a7f Mon Sep 17 00:00:00 2001
From: Sampath Vangaveti <sampathv@codeaurora.org>
Date: Wed, 4 Apr 2018 20:27:40 +0530
Subject: [PATCH 19/19] Camera2Client: Fix issue with AE Bracketing mode.

In Camera driver, new vendor tag added to enable
ae bracket mode. Use the same vendor tag value
while updating capture request settings.

Change-Id: If37979aad369c22f05e88caa23083a7c80730b9b
---
 .../api1/qticlient2/QTIParameters.cpp         | 21 +++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/services/camera/libcameraservice/api1/qticlient2/QTIParameters.cpp b/services/camera/libcameraservice/api1/qticlient2/QTIParameters.cpp
index da670bc17..c98b424a4 100644
--- a/services/camera/libcameraservice/api1/qticlient2/QTIParameters.cpp
+++ b/services/camera/libcameraservice/api1/qticlient2/QTIParameters.cpp
@@ -171,6 +171,9 @@ const char KEY_QTI_AE_BRACKET_HDR[] = "ae-bracket-hdr";
 const char AE_BRACKET_OFF[] = "Off";
 const char AE_BRACKET[] = "AE-Bracket";
 
+const char KEY_QTI_VENDOR_AE_BRACKET_MODE[] =
+        "org.codeaurora.qcamera3.ae_bracket.mode";
+
 // HFR
 const char KEY_QTI_VIDEO_HIGH_FRAME_RATE[] = "video-hfr";
 const char KEY_QTI_VIDEO_HIGH_SPEED_RECORDING[] = "video-hsr";
@@ -1045,8 +1048,16 @@ status_t QTIParameters::updateRequest(CameraMetadata *request) const {
 
 status_t QTIParameters::updateRequestForQTICapture(Vector<CameraMetadata> *requests) const {
     status_t res = OK;
+    uint32_t tag = 0;
     sp<VendorTagDescriptor> vTags =
         VendorTagDescriptor::getGlobalVendorTagDescriptor();
+    if ((nullptr == vTags.get()) || (0 >= vTags->getTagCount())) {
+        sp<VendorTagDescriptorCache> cache =
+                VendorTagDescriptorCache::getGlobalVendorTagCache();
+        if (cache.get()) {
+            cache->getVendorTagDescriptor(vendorTagId, &vTags);
+        }
+    }
 
     if (!requests) {
        return BAD_VALUE;
@@ -1110,6 +1121,16 @@ status_t QTIParameters::updateRequestForQTICapture(Vector<CameraMetadata> *reque
             if (res != OK) {
                 return res;
             }
+
+            //Update ae bracket mode to 1.
+            uint8_t AEBracketMode = 1;
+            res = CameraMetadata::getTagFromName(KEY_QTI_VENDOR_AE_BRACKET_MODE, vTags.get(), &tag);
+            if (res == OK) {
+                res = request.update(tag,&AEBracketMode, 1);
+                if (res != OK) {
+                    return res;
+                }
+            }
         }
     }
 
-- 
2.17.1

